File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/image/ImageJpaRepository.java
package kyonggiyo.persistence.image;

import kyonggiyo.application.image.domain.entity.Image;
import kyonggiyo.application.image.domain.vo.ImageType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface ImageJpaRepository extends JpaRepository<Image, Long> {

    List<Image> findByImageTypeAndReferenceId(ImageType imageType, Long referenceId);

    List<Image> findByImageTypeAndReferenceIdIn(ImageType imageType, List<Long> ids);

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/image/repository/ImageJpaRepository.kt
package kyonggiyo.image.repository

import kyonggiyo.application.image.domain.entity.Image
import kyonggiyo.application.image.domain.vo.ImageType
import org.springframework.data.jpa.repository.JpaRepository

interface ImageJpaRepository : JpaRepository<Image, Long> {

    fun findByImageTypeAndReferenceId(imageType: ImageType, referenceId: Long): List<Image>

    fun findByImageTypeAndReferenceIdIn(imageType: ImageType, ids: List<Long>): List<Image>

}
